frontend/packages/dev-console/src/utils/pipeline-utils.ts

```diff
@@ -350,19 +350,29 @@ export const pipelineRunDuration = (run: PipelineRun): string => {
 };
 
 export const updateServiceAccount = (
-  secretName: string,
+  secretNames: string[],
   originalServiceAccount: ServiceAccountType,
 ): Promise<ServiceAccountType> => {
   const updatedServiceAccount = _.cloneDeep(originalServiceAccount);
-  updatedServiceAccount.secrets = [...updatedServiceAccount.secrets, { name: secretName }];
+  secretNames.map(
```

Comment:
You want to avoid using `.map` when you're not assigning the returned value out of `.map`. If you want a loop, it's `.forEach` that won't require a `return` statement on the method.

I'll leave this to you to clean up. Consider using `.reduce` if you think it'll help.

Docs: [MDN Array Reduce](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/Reduce), [MDN Array Map](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map).

Reply:
using `.forEach` here